### PR TITLE
add function getCloneBaseUrlOrHost in GerritSettings

### PR DESCRIPTION
**Pull request by flo-chevalier**
We need to use the clone base URL instead of HOST for fetch a change from the remote repository.

### CODE REVIEW

File: src/main/java/com/urswolfer/intellij/plugin/gerrit/GerritSettings.java [hunk 1]
```diff
@@ -311,8 +312,11 @@ public String getCloneBaseUrl() {
         return cloneBaseUrl;
     }
 
-
     public void setLog(Logger log) {
         this.log = log;
     }
+
+    public String getCloneBaseUrlOrHost() {
+        return StringUtils.isEmpty(cloneBaseUrl) ? host : cloneBaseUrl;
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/intellij/plugin/gerrit/GerritSettings.java]: Use Strings.isNullOrEmpty here (this guava Strings class is already imported here).

**Review by uwolfer — CHANGES_REQUESTED** [review]
Thanks for your PR! You can update this PR by amending the existing commit, and then force push it to the same remote branch.

**Comment by isotes** [comment]
I think the change looks good. Thanks!

Just out of interest I would be like to know in which scenarios there was a problem. In my setup, there is no problem because `cloneBaseUrl` and `gerritHost` are actually on the same host. However, also in other cases I would expect it to work since afaics `getHost` checks a repository's push URLs which I assume normally includes one based on `cloneBaseUrl`. Perhaps add a note in the commit message or a bug report just to document the circumstances where a problem occurred.

File: src/main/java/com/urswolfer/intellij/plugin/gerrit/GerritSettings.java (hunk 1 above)
Comment (flo-chevalier) [reply src/main/java/com/urswolfer/intellij/plugin/gerrit/GerritSettings.java]: Done

**Comment by flo-chevalier** [comment]
> Just out of interest I would be like to know in which scenarios there was a problem. In my setup, there is no problem because `cloneBaseUrl` and `gerritHost` are actually on the same host.

At work, "host" (web client) and "clone base url" haven't got the same IP.

> I would expect it to work since afaics `getHost` checks a repository's push URLs which I assume normally includes one based on `cloneBaseUrl`.

I thought that too. But it does not work for me.

> Perhaps add a note in the commit message or a bug report just to document the circumstances where a problem occurred.

Done

**Review by uwolfer — APPROVED** [review]
Thanks for your work!

**Comment by flo-chevalier** [comment]
> 
> 
> Thanks for your work!

You're welcome.

**Pull request merged**